api/maven-api-core/src/main/java/org/apache/maven/api/Type.java

```diff
@@ -81,6 +81,24 @@ public interface Type extends ExtensibleEnum {
      */
     String MODULAR_JAR = "modular-jar";
 
+    /**
+     * Artifact type name for a JAR file that can be placed either on the annotation processor class-path
+     * or module-path. The path (classes or modules) is chosen by the plugin, possibly using heuristic rules.
+     */
+    String PROCESSOR = "proc";
```

Comment:
Why not `"processor"` ?

Reply:
I agree. `proc` was used because other types have short names (e.g. "jar") and `proc` is the name of the command-line option. But I usually like better explicit names like "processor".